tools/gather-sysinfo/gather-sysinfo.go

```diff
@@ -44,9 +47,27 @@ func newSnapshotCommand(knitOpts *cmd.KnitOptions) *cobra.Command {
 	// use this to debug container behaviour
 	snap.Flags().IntVar(&opts.sleepTime, "sleep", 0, "amount of seconds to sleep once done, before exit")
 
+	collectMachineinfo()
+
 	return snap
 }
 
+func collectMachineinfo() {
+	err := os.MkdirAll("/etc/machineinfo/", os.ModePerm)
+	if err != nil {
+		panic(err)
+	}
+	outfile, err := os.Create("/etc/machineinfo/machineinfo.json")
+	if err != nil {
+		panic(err)
+	}
+
+	mih := machineinformer.Handle{
+		Out: outfile,
```

Comment:
you need to set the value of `Root` here using `sysfsRoot` from `knitOpts`

Reply:
I don't understand this, what Root is used for?

Reply:
It is used to find the root of the sysfs. The name is probably a bit misleading, the concept is mixed with the mount point.
Normally, this value is `/sys`. But we cannot just hardcode `/sys`, because this tool is meant to be used inside a container, so the host sysfs is usally not mounted on `/sys`, but in `/host-sys` or something like this.

It is passed to `GetRaw` - https://github.com/openshift-kni/debug-tools/blob/main/pkg/machineinformer/informer.go#L61
Which use the value to initialize a `relocatablesysfs` - https://github.com/openshift-kni/debug-tools/blob/main/pkg/machineinformer/informer.go#L63
Which in turn builds the path using `root` as prefix - https://github.com/openshift-kni/debug-tools/blob/main/pkg/machineinformer/relocatablesysfs.go#L79

Easy example: https://github.com/openshift-kni/debug-tools/blob/main/pkg/machineinformer/relocatablesysfs.go#L85